tools/spec-gen-sdk/src/types/PackageData.ts

```diff
@@ -136,10 +136,7 @@ export const getIntegrationBranchName = (context: WorkflowContext, packageName:
 
 // tslint:disable-next-line: max-line-length
 export const getPackageData = (context: WorkflowContext, result: PackageResult, suppressionContentList?: SDKSuppressionContentList): PackageData => {
-  const relativeFolderPath = result.path.find(p => {
-    const segments = p.split(path.sep);
-    return segments.includes('sdk') && !path.extname(p);
-  })!;
+  const relativeFolderPath = result.path.find(p => p.split(path.sep).includes('sdk'))!;
```

Comment:
It seems `typeerror` will still happen with this code. We should report errors if `serviceName` cannot be assigned a valid value.
```suggestion
  const relativeFolderPath = result.path
  ?.filter(p => p) 
  .find(p => p.split(path.sep).includes('sdk') && !path.extname(p)) || '';
```

Reply:
`!path.extname(p)` is what cause failure in C#/ Some directories look like files with extension because they use `.` in them.